lib/bunyan.js

```diff
@@ -8,209 +20,136 @@
-    var formatRegExp = /%[sdj%]/g;
-    format = function format(f) {
-        if (typeof (f) !== 'string') {
-            var objects = [];
-            for (var i = 0; i < arguments.length; i++) {
-                objects.push(inspect(arguments[i]));
-            }
-            return objects.join(' ');
-        }
-
-        var i = 1;
-        var args = arguments;
-        var len = args.length;
-        var str = String(f).replace(formatRegExp, function (x) {
-            if (i >= len)
-                return x;
-            switch (x) {
-                case '%s': return String(args[i++]);
-                case '%d': return Number(args[i++]);
-                case '%j': return fastAndSafeJsonStringify(args[i++]);
-                case '%%': return '%';
-                default:
-                    return x;
-            }
-        });
-        for (var x = args[i]; i < len; x = args[++i]) {
-            if (x === null || typeof (x) !== 'object') {
-                str += ' ' + x;
-            } else {
-                str += ' ' + inspect(x);
-            }
-        }
-        return str;
-    };
+  if (obj === null || obj === undefined) {
+    return obj;
+  } else if (Array.isArray(obj)) {
+    return obj.slice();
+  } else if (typeof obj === 'object') {
+    const copy = Object.assign({}, obj);
+    return copy;
+  }
+  return obj;
 }
 
-
 /**
  * Gather some caller info 3 stack levels up.
  * See <http://code.google.com/p/v8/wiki/JavaScriptStackTraceApi>.
  */
 function getCaller3Info() {
-    if (this === undefined) {
-        // Cannot access caller info in 'strict' mode.
-        return;
-    }
-    var obj = {};
-    var saveLimit = Error.stackTraceLimit;
-    var savePrepare = Error.prepareStackTrace;
-    Error.stackTraceLimit = 3;
-
-    Error.prepareStackTrace = function (_, stack) {
-        var caller = stack[2];
-        if (sourceMapSupport) {
-            caller = sourceMapSupport.wrapCallSite(caller);
-        }
-        obj.file = caller.getFileName();
-        obj.line = caller.getLineNumber();
-        var func = caller.getFunctionName();
-        if (func)
-            obj.func = func;
-    };
-    Error.captureStackTrace(this, getCaller3Info);
-    this.stack;
-
-    Error.stackTraceLimit = saveLimit;
-    Error.prepareStackTrace = savePrepare;
-    return obj;
+  // Cannot access caller info in 'strict' mode.
+  if (this === undefined) return;
+
+  let obj = {};
+  const saveLimit = Error.stackTraceLimit;
+  const savePrepare = Error.prepareStackTrace;
+  Error.stackTraceLimit = 3;
+
+  Error.prepareStackTrace = (_, stack) => {
+    let caller = stack[2];
+    if (sourceMapSupport) caller = sourceMapSupport.wrapCallSite(caller);
+
+    obj.file = caller.getFileName();
+    obj.line = caller.getLineNumber();
+    const func = caller.getFunctionName();
+    if (func) obj.func = func;
+  };
+  Error.captureStackTrace(this, getCaller3Info);
+  // eslint-disable-next-line no-unused-expressions
+  this.stack;
```

Comment:
I believe this is an Object key not a function, so can be safely removed?

Reply:
Strangely, if I remove this line, the test will failed.